### ISSUE TITLE
Community Speaker page has data/location validation issues

### ISSUE THREAD

**Issue by mrlacey** [dotnet-foundation/website#508]
![image](https://user-images.githubusercontent.com/189547/96124465-659fb180-0eeb-11eb-8b70-691901be38ac.png)

Either
- The Netherlands has moved and no-one told me,
- there's some faulty geo-data in the system, or
- the location name resolution when adding a new speaker has a bug

---

I was wrong. It seems there's a street in a town outside London that is called "The Netherlands" and the Azure Maps API thinks this is a more likely result than the country.

Based on https://github.com/dotnet-foundation/website/blob/5646f711fba142b99b98f8cea345704e429f99e4/GeocodeLocations.cs#L49

```ascii
// 20201015135708
// https://atlas.microsoft.com/search/address/json?&subscription-key=XXXXXX-HIDING-MY-KEY-XXXXXX&api-version=1.0&language=en-US&limit=1&query=The+Netherlands

{
  "summary": {
    "query": "the netherlands",
    "queryType": "NON_NEAR",
    "queryTime": 54,
    "numResults": 1,
    "offset": 0,
    "totalResults": 51,
    "fuzzyLevel": 1
  },
  "results": [
    {
      "type": "Street",
      "id": "GB/STR/p0/177159",
      "score": 4.6116523743,
      "address": {
        "streetName": "The Netherlands",
        "municipalitySubdivision": "Croydon",
        "municipality": "Coulsdon",
        "countrySecondarySubdivision": "Surrey",
        "countrySubdivision": "ENG",
        "countrySubdivisionName": "England",
        "postalCode": "CR5",
        "extendedPostalCode": "CR5 1ND, CR5 1NG, CR5 1NJ",
        "countryCode": "GB",
        "country": "United Kingdom",
        "countryCodeISO3": "GBR",
        "freeformAddress": "The Netherlands, Coulsdon",
        "localName": "Coulsdon"
      },
      "position": {
        "lat": 51.30505,
        "lon": -0.14558
      },
      "viewport": {
        "topLeftPoint": {
          "lat": 51.30604,
          "lon": -0.14663
        },
        "btmRightPoint": {
          "lat": 51.30375,
          "lon": -0.14425
        }
      }
    }
  ]
}
```

It looks like there's a need to manually update or override the results from the call to the API.

**Comment by daveaglick** [comment]
You got it - we use the Azure Maps API to do fuzzy place name matching. The lat/lon can be overridden in the profile though by using `Lat` and `Lon` values in the front matter (this just isn't well publicized yet, not sure how best to indicate that on the speaker add page). When those are both provided, we use the explicit values instead of calling Azure Maps to get them.

**Comment by augustoproiete** [comment]
@mrlacey @daveaglick Leaving aside the odd Azure Maps API behavior, shouldn't we encourage speakers to add a more detailed location including their city and state (if applicable), in addition to their country?

For example, if I'm running a .NET user group in Amsterdam and I'm looking for speakers, it would be useful to know speakers that are local or closer vs. the ones that are far away.

**Comment by daveaglick** [comment]
Totally agree a more detailed location should be encouraged. Any thoughts on how to suggest/enforce that? I could check for a comma and error out if there isn't one (under the assumption a comma indicates at least a "region, country" style designation.

**Comment by augustoproiete** [comment]
Yes, enforcing at least one comma assuming a `region, country` style would definitely be a good improvement.

That should catch most issues, at least for speakers outside of the US.

US-based speakers tend to omit their country and write things like `City, State` e.g. `Nashville, TN` (though probably not an issue for the Azure Maps API as I'm sure they recognize that much better than locations outside the US).

Perhaps enforcing two commas would be better to send people in the right direction?

e.g.

| Location | Pass/Fail |
| --- |:---:|
| The Netherlands |  :x: |
| Amsterdam, The Netherlands |  :x: |
| Nashville, TN |  :x: |
| Nashville, TN, USA |  :white_check_mark: |
| Amsterdam,, The Netherlands |  :white_check_mark: |
| Sao Paulo, SP, Brazil | :white_check_mark: |
| Sao Paulo,, Brazil | :white_check_mark: |

---

Of course, alternatively, we could have separate fields (`city`, `region`, `country`) in the YAML instead of `location`, and make `city` and `country` required...
